aesara/tensor/shape.py

```diff
@@ -371,22 +371,34 @@ class SpecifyShape(COp):
     def make_node(self, x, shape):
         if not isinstance(x, Variable):
             x = aet.as_tensor_variable(x)
-        shape = aet.as_tensor_variable(shape)
-        if shape.ndim > 1:
-            raise AssertionError()
-        if shape.dtype not in aesara.tensor.type.integer_dtypes:
-            raise AssertionError()
-        if isinstance(shape, TensorConstant) and shape.data.size != x.ndim:
-            raise AssertionError()
-        return Apply(self, [x, shape], [x.type()])
+        if shape == () or shape == []:
+            shape = np.array(shape, dtype=int)
+        tshape = aet.as_tensor_variable(shape)
+        if tshape.ndim != 1:
+            raise AssertionError(
+                f"The `shape` must be a vector. Got `shape={shape}` instead."
+            )
+        if tshape.dtype not in aesara.tensor.type.integer_dtypes:
+            raise AssertionError(
+                f"The `shape` must be an integer type. Got {tshape.dtype} instead."
+            )
```

Comment:
We can just `cast` the `shape` to `int64`.

Reply:
If an element in a shape vector ever becomes a float, that should be an indication that something is (very) wrong upstream. So silently casting it (if we don't need to) didn't feel very nice.

Reply:
If you're concerned about that, then you can use `aet.lvector.filter_variable`; that will normalize the type and emit an informative error.

Reply:
`filter_variable` doesn't convert the integer types either: `TypeError: Cannot convert Type TensorType(int32, (True,)) (of Variable TensorConstant{(1,) of 2}) into Type TensorType(int64, vector).`